stdlib/public/Darwin/Accelerate/ContiguousCollection.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A collection that supports access to its underlying contiguous storage.
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+public protocol _ContiguousCollection: Collection
```

Comment:
I'm very concerned about the proliferation of these protocols across multiple libraries.

Reply:
Then you need to convince @DougGregor to like my pitch to add `ContiguousCollection` to the standard library :)

Reply:
That's…not a great answer. We (Apple SWE) try not to have copies of the same data structures in the actual Accelerate, Dispatch, CF, Foundation, etc; the overlays should follow the same rules.

Reply:
That's precisely why we want to have the protocol in the standard library?

In particular, Accelerate should not have an upwards dependency on Foundation.

Reply:
I happen to agree that we should have this in the stdlib, but that doesn't mean Accelerate should add this now. *If* Accelerate turns out to be the best place for it, Foundation should adopt that. Meanwhile, I don't want us to have several of these protocols *in-tree* and thus *serving as proposed ABI.*

Reply:
Ok, but what's @FlexMonkey or another API author supposed to do here? Do nothing and wait for the core team to get to considering the issue? Or put up draft API for review that can be updated once a decision is made? There's a substantial number of other issues to hammer out in review while we wait to see what the disposition of these protocols is going to be, and I don't think we should wait for that decision before we get started iterating on other API.

Reply:
The other consideration is that within Accelerate, the usage of this protocol is completely understood to mean a contiguous array of non-bridgeable types, bypassing concerns about lazily bridged arrays. This wouldn't apply if it were in Foundation or the standard library.